plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,153 @@
+local Device = require("device")
+
+if Device.isKindle == nil and Device.isKobo == nil then
+    return { disabled = true }
+end
+
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local CoverImage = WidgetContainer:new{
+    name = 'coverimage',
+    is_doc_only = true,
+    settings_id = 0,
+}
+
+function CoverImage:_enabled()
+    return self.enabled
+end
+
+function CoverImage:_restore()
+    return self.enabled and self.restore
+end
+
+function CoverImage:init()
+    self.cover_image_path = G_reader_settings:readSetting("cover_image_path") or "Cover.png"
+    self.enabled = G_reader_settings:isTrue("cover_image_enabled")
+    self.restore = G_reader_settings:isTrue("cover_image_restore") and self.enabled
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function CoverImage:onCloseDocument()
+    logger.dbg("CoverImage: onCloseDocument")
+    local lfs = require("libs/libkoreader-lfs")
+    -- Try to restore the state before start of KOReader.
+    -- If KOReader gets updated during operation (at least on Android) the backup survives.
+    -- Under normal conditions this should not happen.
+    if self.restore then
+        -- Delete image unconditionally, so if no backup exists, there will be no cover file.
+        -- This is useful on Tolino to use the system sleep image after KOReader exit.
+        if lfs.attributes(self.cover_image_path) then
+            os.remove(self.cover_image_path)
+        end
+        -- Restore backup if it exists, so we can use the last image.
+        -- On Tolino a user defined /sdcard/tolino_others.jgp can be used as screensaver on system sleep.
+        if lfs.attributes(self.cover_image_path ..".bak") then
+            os.rename(self.cover_image_path .. ".bak", self.cover_image_path)
+        end
+    end
+end
+
+function CoverImage:onReaderReady(doc_settings)
+   logger.dbg("CoverImage: onReaderReady")
+   if self.enabled and self.ui and self.ui.document then
+        local lfs = require("libs/libkoreader-lfs")
+        local image = self.ui.document:getCoverPageImage()
+        if image then
+            -- Do not override an existing backup, as this only exists on a crash or abnormal update of KOReader.
+            -- This is very usefull, if an image shall be used as a system screensaver (e.g. on Tolino as /sdcard/suspend_other.jpg)
+            if not lfs.attributes(self.cover_image_path ..".bak") then
+                os.rename(self.cover_image_path, self.cover_image_path .. ".bak" )
+            end
+            Device.screen.bb.writePNG(image, self.cover_image_path, false)
```

Comment:
Instead of "Device.screen.bb.writePNG(image, self.cover_image_path, false)" you can use "image:writePNG(self.cover_image_path, false)", so writePNG does not need to be changed in base/ffi/blitbuffer.lua.

Reply:
@elvvis: As you can see this PR just adds two files. For this PR nothing else has to be changed. So I don't understand your comment about base/ffi/blitbuffer.lua

Reply:
`getCoverPageImage()` returns a BB, `writePNG` is a BB method.

You can just do `image:writePNG(self.cover_image_path)` ;).

Reply:
@NiLuJe: Ahh, now I see :-)